src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp

```diff
@@ -201,6 +202,25 @@ QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap
   else
     settings.flags = settings.flags & ~QgsLayoutRenderContext::FlagAntialiasing;
 
+  QVector< double > mapScales = layout->project()->viewSettings()->mapScales();
```

Comment:
Could you move this to a method in QgsLayoutUtils to avoid the duplicate code here and in QgsLayoutDesignerDialog::predefinedScales()?